src/test/java/gov/healthit/chpl/aqa/stepDefinitions/CMSIDsLookupSteps.java

```diff
@@ -61,30 +52,31 @@ public void lookUpCmsId(final String cmsId) {
     public void addListingsToWidget(final String chplIds) {
         String[] ids = chplIds.split(";");
         for (String id : ids) {
-            driver.get(url + "#/product/" + id);
-            WebDriverWait wait = new WebDriverWait(driver, TIMEOUT);
-            wait.until(ExpectedConditions.visibilityOf(ListingDetailsPage.listingName(driver)));
-            ListingDetailsPage.cmsWidgetButton(driver, id).click();
+            getDriver().get(getUrl() + "#/product/" + id);
+            WebDriverWait wait = new WebDriverWait(getDriver(), TIMEOUT);
+            wait.until(ExpectedConditions.visibilityOf(ListingDetailsPage.listingName(getDriver())));
+            ListingDetailsPage.cmsWidgetButton(getDriver(), id).click();
         }
     }
 
     /**
      * Generate a CMS ID.
+     * @throws Exception
      */
     @When("^I generate a CMS ID$")
     public void generateCmsId() {
-        WebDriverWait wait = new WebDriverWait(driver, TIMEOUT);
-        CMSidReverseLookupPage.generateCmsIdButton(driver).click();
-        wait.until(ExpectedConditions.visibilityOf(CMSidReverseLookupPage.cmsIdResults(driver)));
-    }
+       ((JavascriptExecutor) getDriver()).executeScript("arguments[0].removeAttribute('disabled','disabled')", CMSidReverseLookupPage.generateCmsIdButton(getDriver()));
```

Comment:
Whoa!

This looks like a step that's making an element clickable when the page doesn't think it should be. That seems...improper

Reply:
I can see that being a bad practice in general but in this specific case, I know the element should be 'enabled'. I tested with isEnabled() method which returned false. 'Get cert ID' button was displayed but not enabled and it won't receive a click and the assert was failing as ID won't generate. If it can be enabled with some update, I'll change the test back to simple click event.

Reply:
Agree that it should be clickable, but my guess is that the system is checking to see that the supplied Listings are enough to generate a CMS ID. The test should have a Wait condition until the item is clickable, to allow the system to process the addition of that listing and let the user know that it makes up a complete CMS-valid set of listings

Reply:
If something changed in the validation logic for a CMS id and this set of listings became not-valid for a CMS id (or one of the listings was edited in such a way to make it invalid) we'd want this test to not be able to click, because it wouldn't be valid

Reply:
I did try wait before click event, it didn't work with even the explicit wait. The issue was/is with visibility condition and not load time. Let me look into alternate ways to locate and access this element.

Reply:
If it's a visibility issue, not a clickable issue, then you should be able to use the javascript.execute(something-something[0].click()) thing

Reply:
Tried that, didn't help